assets/js/checkout.js

```diff
@@ -0,0 +1,222 @@
+      $checkout.removeAttr('data-cb-addons_id_1');
+      $checkout.removeAttr('data-cb-addons_quantity_1');
+    }
+
+    if (checkoutOptions.setup_deployment) {
+      $checkout.attr({
+        'data-cb-addons_id_3': 'chargebee-refarch'
+      });
+    } else {
+      $checkout.removeAttr('data-cb-addons_id_3');
+    }
+  }
+
+  function _calculatePrice() {
+    if (checkoutOptions.enterprise) return; // Don't calculate on enterprise
+
+    var total, subtotal, subscriptionTotal, additionalUsers = checkoutOptions.users > 5 ? checkoutOptions.users - 5 : 0;
+
+    if (checkoutOptions.dedicated_support) {
+      if (additionalUsers > 0) total = subtotal = pricing.tier1.price[1] + (additionalUsers * pricing.tier2.price[1]);
+      else total = subtotal = pricing.tier1.price[1]; // 5 or less users
+
+    } else {
+      // Without dedicated support
+      if (additionalUsers > 0) total = subtotal = pricing.tier1.price[0] + (additionalUsers * pricing.tier2.price[0]);
+      else total = subtotal = pricing.tier1.price[0]; // 5 or less users
+    }
+
+    if (checkoutOptions.setup_deployment) {
+      subscriptionTotal = subtotal;
+      subtotal += 4950;
+    }
+
+    $('#subscription-price').text(total.toLocaleString());
+    $('#subscription-subtotal').text(subtotal.toLocaleString());
+
+    _deferCheckout(checkoutOptions.users, checkoutOptions.dedicated_support, checkoutOptions.setup_deployment);
+  }
+
+  // Prevents spamming Chargebee registerAgain on every change
+  function _deferCheckout(users, support, setup) {
+    var cbInstance;
+    if (typeof timeout !== 'undefined') clearTimeout(timeout);
+    $checkout.attr('disabled', true).text('Please wait...');
+
+    timeout = setTimeout(function () {
+      $checkout.attr('disabled', false).text('Checkout');
+      clearTimeout(timeout);
+      //_updateAttrs();
+      Chargebee.registerAgain();
+      cbInstance = Chargebee.getInstance();
+      function htmlEncode(value){
+        if (value) {
+          return jQuery('<div />').text(value).html();
+        } else {
+          return '';
+        }
+      }
+      cbInstance.setCheckoutCallbacks(function(cart, product) {
+        var subscriptionDetails = ("Gruntwork Subscribers: " + users);
+        if (support){
+          subscriptionDetails += " • Dedicated Support";
+        }
+        if (setup){
+          subscriptionDetails += " • Reference Architecture";
+        }
+        //console.log(subscriptionDetails);
+        // you can define a custom callbacks based on cart object
+        var customer = {cf_subscription_details: subscriptionDetails};
+
+        cart.setCustomer(customer);
+
+        return {
+          close: function() {
+            // Required to remove overflow set by the modal
+            // Same as: https://lodash.com/docs/4.17.10#debounce
+            setTimeout(function() { $body.removeAttr('style'); }, 0);
+          }
+        }
+      });
+    }, 1250);
+  }
+
+  _updateCheckout();
```

Comment:
Code looks reasonable... But I'd be terrified of changing it. There's lots more logic here in our own page than I realized!

Can we add automated tests around this? At the minimum, unit tests for the various functions, but ideally, integration tests that use Chargebee in a checkout flow too. Otherwise, the very next update is very likely to break this flow... And that's obviously a problem.